projects/demo-integrations/src/tests/kit/date-time/date-time-time-step.cy.ts

```diff
@@ -0,0 +1,145 @@
+                    .type('213212.')
+                    .should('have.value', '22.12;21:32:12.')
+                    .should('have.a.prop', 'selectionStart', '22.12;21:32:12.'.length)
+                    .should('have.a.prop', 'selectionEnd', '22.12;21:32:12.'.length)
+                    .type(`${'{upArrow}{rightArrow}'.repeat(3)}{downArrow}`)
+                    .should('have.value', '22.12;21:32:12.002')
+                    .should('have.a.prop', 'selectionStart', '22.12;21:32:12.002'.length)
+                    .should('have.a.prop', 'selectionEnd', '22.12;21:32:12.002'.length);
+            });
+
+            it('type 213212. => 22.12;21:32:12.| => type ({downArrow}{rightArrow}) * 3 + {downArrow} => 22.12:21:32:12.995|', () => {
+                cy.get('@input')
+                    .type('213212.')
+                    .should('have.value', '22.12;21:32:12.')
+                    .should('have.a.prop', 'selectionStart', '22.12;21:32:12.'.length)
+                    .should('have.a.prop', 'selectionEnd', '22.12;21:32:12.'.length)
+                    .type(`${'{downArrow}{rightArrow}'.repeat(3)}{downArrow}`)
+                    .should('have.value', '22.12;21:32:12.996')
+                    .should('have.a.prop', 'selectionStart', '22.12;21:32:12.996'.length)
+                    .should('have.a.prop', 'selectionEnd', '22.12;21:32:12.996'.length);
+            });
+
+            it('should affect only time segments', () => {
+                cy.get('@input')
+                    .type('123456111')
+                    .should('have.value', '22.12;12:34:56.111')
+                    .should('have.a.prop', 'selectionStart', '22.12;12:34:56.111'.length)
+                    .should('have.a.prop', 'selectionEnd', '22.12;12:34:56.111'.length)
+                    .type('{upArrow}{leftArrow}'.repeat('22.12;12:34:56.111'.length))
+                    .should('have.value', '22.12;15:37:59.115')
+                    .should('have.a.prop', 'selectionStart', 0)
+                    .should('have.a.prop', 'selectionEnd', 0);
+            });
+        });
+
+        describe('timeStep = 0', () => {
+            beforeEach(() => {
+                cy.visit(
+                    `/${DemoPath.DateTime}/API?dateTimeSeparator=;&dateMode=yy%2Fmm&timeStep=0&timeMode=HH:MM:SS.MSS`,
+                );
+                cy.get('#demo-content input')
+                    .should('be.visible')
+                    .first()
+                    .focus()
+                    .as('input');
+
+                cy.get('@input')
+                    .type('1202123456000')
+                    .should('have.value', '12.02;12:34:56.000')
+                    .should('have.a.prop', 'selectionStart', '12.02;12:34:56.000'.length)
+                    .should('have.a.prop', 'selectionEnd', '12.02;12:34:56.000'.length);
+            });
+
+            it('should be disabled', BROWSER_SUPPORTS_REAL_EVENTS, () => {
+                cy.get('@input').realPress('ArrowUp');
```

Comment:
Let's use the built-in `{upArrow}` ?

Reply:
We use `realPress` only for creating selection via keyboard arrows

Reply:
We cant use built in, because it wont trigger browser event